trpcultivate_phenotypes/tests/src/Kernel/Services/Traits/ValidTraitTest.php

```diff
@@ -200,12 +222,140 @@ public function testTraitsServiceDatabaseExpectations() {
+      'use with Tripal Cultivate Phenotypes. To configure this genus, go to ' .
+      Url::fromRoute('trpcultivate_phenotypes.settings_ontology')->toString() .
+      ' and set the controlled vocabularies associated with this genus.',
+      $exception_caught,
+      "We expected an exception to be caught for this scenario, but one wasn't thrown.",
+    );
+
+    // Test case for where the name is not returned.
+    $mock_statement = $this->getMockBuilder(StatementWrapperIterator::class)
+      ->disableOriginalConstructor()
+      ->onlyMethods(['fetchField'])
+      ->getMock();
+    $mock_statement->method('fetchField')
+      ->willReturn(NULL);
+
+    $mock_database = $this->getMockBuilder(ChadoConnection::class)
+      ->disableOriginalConstructor()
+      ->onlyMethods(['query'])
+      ->getMock();
+    $mock_database->method('query')
+      ->willReturn($mock_statement);
+
+    $this->container->set('tripal_chado.database', $mock_database);
+
+    $this->container->set('trpcultivate_phenotypes.traits', NULL);
+    $this->service_traits = \Drupal::service('trpcultivate_phenotypes.traits');
+
+    $exception_message = 'NONE';
+    $exception_message = '';
+    try {
+      $this->service_traits->setTraitGenus($this->genus);
+    }
+    catch (\Exception $e) {
+      $exception_message = $e->getMessage();
+    }
+    $this->assertStringStartsWith(
+      'We were unable to retrieve the name for the Genus ',
+      $exception_message,
+      "The expected error was not thrown.",
+    );
+
+    // Test case where the terms are not set correctly.
+    $new_terms = [
+      'method_to_trait_relationship_type' => 0,
+      'unit_to_method_relationship_type' => -2,
+    ];
+    $this->config = \Drupal::configFactory()->getEditable('trpcultivate_phenotypes.settings');
+    foreach ($new_terms as $key => $value) {
+      $this->config->set("trpcultivate.phenotypes.ontology.terms.$key", $new_terms[$key]);
+    }
+
+    $this->container->set('trpcultivate_phenotypes.traits', NULL);
+    $this->service_traits = \Drupal::service('trpcultivate_phenotypes.traits');
+    $exception_caught = FALSE;
+    $exception_message = 'NONE';
+    $exception_message = '';
+    try {
+      $this->service_traits->setTraitGenus($this->genus);
+    }
+    catch (\Exception $e) {
+      $exception_caught = TRUE;
+      $exception_message = $e->getMessage();
+    }
+    $this->assertEquals(
+      'Exception: Term(s) method_to_trait_relationship_type, unit_to_method_relationship_type used to create trait
+      asset relationships was not configured. To configure terms, go to' . Url::fromRoute('trpcultivate_phenotypes.settings_ontology')->toString() .
+      'and set the controlled vocabulary associated with the term.',
+      $exception_caught,
+      "We expected an exception to be caught for this scenario, but one wasn't thrown.",
+    );
+  }
+
+  /**
+   * Test the insertTrait() method.
+   */
+  public function testInsertTrait() {
```

Comment:
This test looks like it's only triggering an exception when a genus hasn't been set, but the name implies it's testing the whole method. I would suggest renaming to something such as `testInsertTraitUnsetGenus()` and update the description accordingly.